dapp/src/utils/utils.js

```diff
@@ -33,3 +36,52 @@ export function mergeDeep(target, ...sources) {
 
   return mergeDeep(target, ...sources)
 }
+
+/**
+ * Get stable coin logs
+ */
+
+export const getStableCoinLogs = async (transactions) => {
+  const data = {
+    usdt: 0,
+    dai: 0,
+    usdc: 0,
+    ousd: 0,
+  }
+  const stableCoins = [
+    {
+      address: addresses.mainnet.USDT.toLowerCase(),
+      name: 'usdt',
+      decimals: 6,
+    },
+    {
+      address: addresses.mainnet.USDC.toLowerCase(),
+      name: 'usdc',
+      decimals: 6,
+    },
+    {
+      address: addresses.mainnet.DAI.toLowerCase(),
+      name: 'dai',
+      decimals: 18,
+    },
+    {
+      address: addresses.mainnet.OUSDProxy.toLowerCase(),
+      name: 'ousd',
+      decimals: 18,
+    },
+  ]
+
+  await Promise.all(
+    stableCoins.map((coin) => {
+      const log = transactions.logs.find(
+        (log) => log.address.toLowerCase() === coin.address
+      )
+      if (log) {
+        const value = ethers.BigNumber.from(log.data)
+        data[coin.name] = (parseFloat(value.toString()) / 10) * coin.decimals
```

Comment:
there is a [formatCurrency](https://github.com/OriginProtocol/origin-dollar/blob/6e2b1738fe876aa8e7e6d38c23417383ed33b3ad/dapp/src/utils/math.js#L29) helper function that you can use to format values.